### ISSUE TITLE
Haddock source not generated with GHC 7.10.1

### ISSUE THREAD

**Issue by filib** [filib/ruby-marshal#50]
I've found that this works properly on my work machine which has a different version of GHC, Cabal and Haddock.

## Work Machine

```
[filib@benbecula:~/Projects/ruby-marshal on master]
% ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.8.4
[filib@benbecula:~/Projects/ruby-marshal on master]
% cabal --version
cabal-install version 1.22.6.0
using version 1.22.4.0 of the Cabal library
[filib@benbecula:~/Projects/ruby-marshal on master]
% haddock --version
Haddock version 2.14.3, (c) Simon Marlow 2006
Ported to use the GHC API by David Waern 2006-2008
```

## Personal Machine

```
[filib@colonsay:~/Projects/ruby-marshal on issue-49]
% ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.10.1
[filib@colonsay:~/Projects/ruby-marshal on issue-49]
% cabal --version
cabal-install version 1.22.2.0
using version 1.22.2.0 of the Cabal library
[filib@colonsay:~/Projects/ruby-marshal on issue-49]
% haddock --version
Haddock version 2.16.0, (c) Simon Marlow 2006
Ported to use the GHC API by David Waern 2006-2008
```

**Comment by filib** [comment]
Reported upstream.

**Comment by filib** [comment]
This appears to have been fixed by updating `cabal-install` to 1.22.6.0 and ensuring `hscolour` was on my `PATH`. I think it's a real shame that `cabal-install` doesn't fail more loudly when I explicitly request `--hyperlink-source` and `hscolour` isn't on my `PATH`.
